src/libostree/ostree-repo.c

```diff
@@ -678,7 +678,7 @@ ostree_repo_auto_lock_push (OstreeRepo          *self,
   if (!ostree_repo_lock_push (self, lock_type, cancellable, error))
     return NULL;
 
-  OstreeRepoAutoLock *auto_lock = g_new0 (OstreeRepoAutoLock, 1);
+  OstreeRepoAutoLock *auto_lock = g_slice_new (OstreeRepoAutoLock);
```

Comment:
Out of curiosity, does this still matter today?  I thought modern malloc had subsumed the optimizations here.

Reply:
I honestly have no idea. It seemed like a good idea, but maybe this is based on 2012 knowledge. I also wrote a stack allocated version in https://github.com/dbnicholson/ostree/commit/bd0fb98597ec29a839f67cb75416107368b9cb48 as discussed in one of the comments above.

Reply:
> I thought modern malloc had subsumed the optimizations here.

That is correct. The performance gap between the two is a lot narrower now, and gslice is basically unmaintained. We are (still) [looking at dropping it](https://gitlab.gnome.org/GNOME/glib/-/issues/1079). It’s fine to keep using it, but it might secretly become a call to `g_malloc()` in future.

Reply:
Good to know! I'll file a followup to change back `g_new`.

Reply:
Not really worth the churn, imo. They both work fine; I prefer `g_malloc()`+friends for new code, but am not going around changing old code from one to the other.

Reply:
Any way you slice it, OK by me.